### DISCUSSION TITLE
Database tables prefix

### DISCUSSION THREAD

**Discussion by FahedAlkaabi** [vanilophp/framework#187] [Ideas]
Thank you for great script, could you add database tables prefix on concord.php

I did simple solution and it works but it could be faster if you add prefix element on concord

> app/Models/Product.php
```<?php

namespace App\Models;

use Vanilo\Product\Models\Product as VaniloProduct;

class Product extends VaniloProduct
{
    protected $table = 'eco_products';  // Set the table name with the "eco_" prefix
}
```

> config/concord.php

```
return [
    'modules' => [
        Vanilo\Product\Providers\ModuleServiceProvider::class => [
            'models' => [
                Vanilo\Product\Contracts\Product::class => App\Models\Product::class,  // Register custom Product model
            ],
        ],
        Vanilo\Order\Providers\ModuleServiceProvider::class => [
            'models' => [
                Vanilo\Order\Contracts\Order::class => App\Models\Order::class,  // Register custom Order model
            ],
        ],
        // Add other models here as needed
    ]
];
```

**Comment by fulopattila122** [comment]
The custom models can be registered in the application's `AppServiceProvider`:

```php
class AppServiceProvider extends ServiceProvider
{
    public function boot()
    {
        $this->app->concord->registerModel(Vanilo\Product\Contracts\Product::class, App\Models\Product::class);
        $this->app->concord->registerModel(Vanilo\Order\Contracts\Order::class => App\Models\Order::class);
    }
}
```

https://vanilo.io/docs/4.x/models#using-custom-models

Or do you mean a new feature to set a prefix globally on every table?

**Reply by FahedAlkaabi** [reply]
Could be write all models should be added, I tried but I couldn’t find dependencies of migrations

**Reply by fulopattila122** [reply]
> Could be write all models should be added, I tried but I couldn’t find dependencies of migrations

Sorry I don't understand that

**Comment by FahedAlkaabi** [comment]
I mean globally for every table, I think concord have solutions about this

**Reply by fulopattila122** [reply]
I'm not aware Concord having support for table prefixes, but it could be a nice feature.
